crates/tauri-cli/src/interface/rust/desktop.rs

```diff
@@ -304,16 +305,26 @@ fn build_command(
     args.extend(options.args);
   }
 
+  let has_custom_profile = args.contains(&"--profile".to_string());
+
   let mut features = config_features;
   if let Some(f) = options.features {
     features.extend(f);
   }
+  // custom-protocol is handled in tauri's build.rs script
+  // which makes `cargo build` same as `tauri dev`
+  // and `cargo build --release` same as `tauri build`
+  // but for `tauri build --debug` we need to build the code
+  // in `debug` profile and activate `custom-protocol` feature.
+  if !dev && options.debug && !has_custom_profile {
```

Comment:
```suggestion
  if !dev {
```
if the custom profile is a debug one we still want to force the feature flag

Reply:
[`3de0343` (#11098)](https://github.com/tauri-apps/tauri/pull/11098/commits/3de0343d5770fea0ecf61d3dc985eb36e6d63503)